### PR TITLE
[mdesktopentry] Don't log boolean lookup errors

### PR DESCRIPTION
**Pull request by thp**

### CODE REVIEW

**Comment by pvuorela** [comment]
LGTM, but while at it, could you make rest of the methods defined to have fallback value not to warn. url(), categories(), etc.

**Comment by thp** [comment]
PR updated, now all direct accessor functions have fallback values.

File: src/mdesktopentry.cpp [hunk 1]
```diff
@@ -379,11 +383,16 @@ uint MDesktopEntry::hash() const
 
 QString MDesktopEntry::type() const
 {
+    // Type always has to be present (see isValid())
```
Comment (pvuorela) [inline src/mdesktopentry.cpp]: Being strict, desktop file and isValid() require this, but the API doesn't. But suppose that's irrelevant.

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**